Content/Buffs/Hexes/Hex.cs

```diff
@@ -122,13 +122,14 @@ public override void UpdateLifeRegen(NPC npc, ref int damage)
                 hex.modHex.UpdateLifeRegen(ref damage);
             }
         }
-        public override bool PreDraw(NPC npc, SpriteBatch spriteBatch, Color drawColor)
+
+        public override bool PreDraw(NPC npc, SpriteBatch spriteBatch, Vector2 screenPos, Color drawColor)
```

Comment:
This one is pretty important. NPC PreDraw/PostDraw now get a `Vector2 screenPos` argument, which should be used instead of Main.screenPosition. If weird stuff happens when NPCs are drawn in the bestiary, this is almost certainly why.